tests/derive/test_assign_school_year.py

```diff
@@ -9,30 +9,42 @@
+            ("Wales",               "09", "01",     "09", "01"),
+            ("Scotland",            "08", "15",     "03", "01"),
+            ("Northern Ireland",    "09", "01",     "07", "02"),
+            # fmt: on
         ],
         schema=["country", "school_start_month", "school_start_day", "school_year_ref_month", "school_year_ref_day"],
     )
 
 
 @pytest.mark.parametrize(
     "expected_data",
+    # fmt: off
     [
-        ("2021-11-10", "2014-09-02", "England", 2),
-        ("2021-11-10", "2014-09-02", "Northern Ireland", 2),
-        ("2021-11-10", "2014-09-02", "Scotland", 2),
-        ("2021-08-15", "2014-09-02", "Wales", 1),
-        ("2021-08-15", "2014-09-02", "Northern Ireland", 1),
-        ("2021-08-15", "2014-09-02", "Scotland", 2),
-        ("2021-09-01", "2014-07-01", "Northern Ireland", 3),
-        ("2021-08-14", "2014-02-28", "Scotland", 2),
-        ("2021-08-14", "1995-02-28", "Scotland", None),
-        ("2021-11-10", "2016-10-01", "England", None),
-        ("2021-11-10", "2016-10-01", "Northern Ireland", None),
+        ("2021-09-01", "2014-09-01", "England",             2),
+        ("2021-08-31", "2014-09-01", "England",             1), # new reference cutoff not started and child born after start date
+        ("2021-09-01", "2014-08-31", "England",             3), # new reference cutoff started and child born before start date
+        ("2021-08-31", "2014-08-31", "England",             2), # new reference cutoff not started and child born after before date
+        ("2021-09-01", "2017-08-31", "England",             0),
+        ("2021-09-01", "2017-09-01", "England",             None),  # too young to start school
+
+        ("2021-08-15", "2014-08-15", "Scotland",            2), # testing edge case after school start
+        ("2021-08-14", "2014-08-15", "Scotland",            1), # new school year not yet started and child born after start date
+        ("2021-08-15", "2014-08-14", "Scotland",            2), # new school year started and child born before start date
+        ("2021-08-14", "2014-08-14", "Scotland",            1), # new school year not yet started and child born before start date
+
+        ("2021-03-01", "2014-03-01", "Scotland",            1), # testing edge case after date allocation cut off
+        ("2021-02-28", "2014-03-01", "Scotland",            1), # new reference cutoff not started and child born after start date
+        ("2021-03-01", "2014-02-28", "Scotland",            2), # new reference cutoff started and child born before start date
+        ("2021-02-28", "2014-02-28", "Scotland",            2), # new reference cutoff not started and child born after before date
+
+        ("2021-08-15", "2014-02-28", "Scotland",            3), # new school year started and child born before reference cutoff date
+        ("2021-08-15", "2014-03-01", "Scotland",            2), # new school year started and child born after reference cutoff date
     ],
```

Comment:
Add additional check (with Wales as the country) to check max school year (i.e. 13 max, above 13 should be null). Wales being the country only so then all countries are included which will double check the lookup

Reply:
is the maximum year 12 instead of 13 @osborj ?